Added config dependency provider

vmikki opened the pull request.

mattbennett commented on test/test_dependencies.py:
```diff
@@ -0,0 +1,20 @@
+    @pytest.yield_fixture(autouse=True)
+    def patch_container(self):
+        with patch.object(Config, 'container'):
+            yield
```
This is quite an "internal" looking test. For such a simple dependency provider it would be better just to demonstrate that it works.

You could simply duplicate the test you have above to verify your example. That is "real-world" usage, therefore less brittle, and would show an example service and how to use it next to each other.

vmikki commented:
Thank for the review! I have addressed your comments and modified the example so it more looks like a "real-life usage". I have also added myself to the contributors.

davidszotten commented on docs/examples/test/test_examples.py:
```diff
@@ -281,3 +282,31 @@ def test_websocket_rpc(self, container_factory, web_config, websocket):
+        container.start()
+
+        with entrypoint_hook(container, "foo") as foo:
+            assert "foo" == foo()
```
we normally avoid [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) in python (here and below)

mattbennett replied:
I didn't know this had a name!

vmikki replied:
I came from PHP, it's good to know that it should be avoided in python, thanks!

davidszotten replied:
it's not harmful, but it reads a bit odd and doesn't add anything, since python uses different operators for comparison and assignment, and e.g. `if a = 2:` is a syntax error

davidszotten commented:
may be worth a note that this is _read only_ access to the config

mattbennett commented:
Good shout. I'll add this in the RC branch
